core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -526,16 +544,8 @@ private[spark] class BlockManager(
 
     blockManagerId = if (idFromMaster != null) idFromMaster else id
 
-    shuffleServerId = if (externalShuffleServiceEnabled) {
-      logInfo(s"external shuffle service port = $externalShuffleServicePort")
-      BlockManagerId(executorId, blockTransferService.hostName, externalShuffleServicePort)
-    } else {
-      blockManagerId
-    }
```

Comment:
This could be remained unchanged? If so, I think it's better to keep `shuffleServerId` assignment in one place.

Reply:
When ESS is enabled, `shuffleServiceId` should be assigned before calling `registerWithExternalShuffleServer()` because that method uses it.
When ESS is not enabled, then `shuffleServiceId` is the id with topology and I didn't want to change that.